internal_packages/thread-list/lib/thread-list.cjsx

```diff
@@ -284,13 +284,18 @@ class ThreadList extends React.Component
 
   _onRemoveFromView: =>
     threads = @_threadsForKeyboardAction()
+    backspaceDelete = NylasEnv.config.get('core.reading.backspaceDelete')
     if threads
-      if FocusedMailViewStore.mailView().canArchiveThreads()
-        removeMethod = TaskFactory.taskForArchiving
-      else if FocusedMailViewStore.mailView().canTrashThreads()
-        removeMethod = TaskFactory.taskForMovingToTrash
+      if backspaceDelete
+        if FocusedMailViewStore.mailView().canTrashThreads()
+          removeMethod = TaskFactory.taskForMovingToTrash
+        else
+          return
       else
-        return
+        if FocusedMailViewStore.mailView().canArchiveThreads()
+          removeMethod = TaskFactory.taskForArchiving
+        else
+          return
```

Comment:
Can we fall back to `removeMethod = TaskFactory.taskForMovingToTrash` on this line? I think that since backspaceDelete is false by default, users who don't have an archive folder will reach here, `canArchiveThreads` will return false, and nothing will happen when they press "delete" :-)